src/System.Private.CoreLib/shared/System/Runtime/Serialization/SerializationInfo.cs

```diff
@@ -0,0 +1,514 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Runtime.Serialization
+{
+    /// <summary>The structure for holding all of the data needed for object serialization and deserialization.</summary>
+    public sealed class SerializationInfo
+    {
+        private const int DefaultSize = 4;
+
+        // Even though we have a dictionary, we're still keeping all the arrays around for back-compat. 
+        // Otherwise we may run into potentially breaking behaviors like GetEnumerator() not returning entries in the same order they were added.
+        private string[] _names;
+        private object[] _values;
+        private Type[] _types;
+        private int _count;
+        private Dictionary<string, int> _nameToIndex;
+        private IFormatterConverter _converter;
+        private string _rootTypeName;
+        private string _rootTypeAssemblyName;
+        private Type _rootType;
+
+        [CLSCompliant(false)]
+        public SerializationInfo(Type type, IFormatterConverter converter) 
```

Comment:
Nit: This is not the CoreRT version ... the CoreRT version is:

```
        [CLSCompliant(false)]
        public SerializationInfo(Type type, IFormatterConverter converter, bool requireSameTokenInPartialTrust)
            : this(type, converter)
        {
            // requireSameTokenInPartialTrust is a vacuous parameter in a platform that does not support partial trust.
        }
```

Reply:
I just moved this lower down on line 53